### PR TITLE
[PLAT-10654] Add --dry-run flag to upload commands

### PR DESCRIPTION
**Pull request by joshedney**
## Goal

Validate but not upload source maps for all supported languages.

## Changeset

Add the `--dry-run` flag to all upload commands to process given source maps but not upload them. 
Removed the dry run log message from the `AAB` command to reduce the number of log messages

## Testing

Tested manually, CI tests to be added during testing rework.

### CODE REVIEW

File: pkg/upload/android-proguard.go [hunk 1]
```diff
@@ -27,6 +27,10 @@ func ProcessAndroidProguard(apiKey string, applicationId string, appManifestPath
 	var appManifestPathExpected string
 	var err error
 
+	if dryRun {
+		log.Info("Performing dry run - no files will be uploaded")
+	}
```
Comment (tomlongridge) [inline pkg/upload/android-proguard.go:32]: Does dryRun now apply to all commands? Could we just put a message at the very start, top-level? I guess it just needs to be worded more generally – "no data will be sent to BugSnag"?

File: pkg/server/request.go [hunk 2]
```diff
@@ -69,32 +66,37 @@ func SendRequest(request *http.Request, timeout int) (*http.Response, error) {
...
+
+		if !statusOK {
+			return fmt.Errorf("%s : %s", res.Status, string(b))
+		}
 
-	if !statusOK {
-		return fmt.Errorf("%s : %s", res.Status, string(b))
+		log.Success("Uploaded " + filepath.Base(fileName))
+	} else {
+		log.Success("(dryrun) Uploaded " + filepath.Base(fileName))
```
Comment (lemnik) [inline pkg/server/request.go:99]: I would still be nervous about this wording - "Uploaded" still sounds like something happened.

I would personally prefer something that made it absolutely clear nothing had happened. Something like:

```go
log.Success("(dryrun) Skipping upload of " + filepath.Base(fileName) + " to " + endpoint)
```

Comment (joshedney) [reply pkg/server/request.go:99]: Thoughts @tomlongridge ?

Comment (tomlongridge) [reply pkg/server/request.go:99]: Agreed - maybe also `.Info` log to underscore it?

Are you going to do equivalent for `Create.go`?

**Review by tomlongridge — APPROVED** [review]

**Pull request merged**